Gems/AzQtComponentsForPython/CMakeLists.txt

```diff
@@ -0,0 +1,224 @@
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+# Define the shared library for which we will create bindings.
+set_property(TARGET ${AZQTCOMPONENTS_LIBRARY} PROPERTY PREFIX "")
+
+# Ensure the generated wrapper files have the correct header.
+set(LICENSE_FILE ${CMAKE_CURRENT_LIST_DIR}/license_header.txt)
+
+# Set up the options to pass to shiboken.
+set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
+    --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero
+    --avoid-protected-hack --language-level=c++17
+    --license-file=${LICENSE_FILE}
+    ${INCLUDES}
+    -I${CMAKE_SOURCE_DIR}
+    -I${CMAKE_CURRENT_LIST_DIR}/..
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzCore"
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzQtComponents"
+    -I"${PROJECT_BINARY_DIR}/../AzQtComponents_autogen/include"
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzCore/Platform/${PAL_PLATFORM_NAME}"
+    -I"${PYSIDE2_INCLUDE_DIR}"
+    -I"${SHIBOKEN_INCLUDE_DIR}"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtWidgets"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtCore"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtGui"
+    -I"${PYSIDE2_TYPESYSTEM_PATH}"   
+    -I"${QT_MODULE_DIR}/include"
+    -I"${QT_MODULE_DIR}/include/QtWidgets"
+    -I"${QT_MODULE_DIR}/include/QtCore"
+    -I"${QT_MODULE_DIR}/include/QtGui"
+    -T"${PYSIDE2_PATH}/typesystems"
+    -T"${PYSIDE2_PATH}"
+    -T"${SHIBOKEN_INCLUDE_DIR}"
+    -T"${PYSIDE2_TYPESYSTEM_PATH}"
+    -T"${PYSIDE2_SHARE_PATH}"
+    --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+)
+
+set(generated_sources_dependencies ${wrapped_header} ${typesystem_file})
+
+# Dummy library to run the custom shiboken command.
+ly_add_target(
+    NAME ${BINDINGS_LIBRARY}.bindings STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        dummy_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            ${CMAKE_CURRENT_LIST_DIR}/..
+            ${PYSIDE2_INCLUDE_DIR}
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzQtComponents
+            Gem::EditorPythonBindings.Static
+)
+
+# Add the custom shiboken command to the dummy library.
+add_custom_command(TARGET ${BINDINGS_LIBRARY}.bindings
+                    POST_BUILD
+                    COMMAND ${SHIBOKEN_GENERATOR_EXE_PATH} ${shiboken_options} ${wrapped_header} ${typesystem_file}
+                    DEPENDS ${generated_sources_dependencies} 
+                        Gem::AzQtComponentsForPython.WrappedFiles
+                    COMMENT "Running generator for ${typesystem_file}.")
+    
+# Create dummy wrapper files so the second project has something to work with while being created.
+include(wrapped_files.cmake)
+foreach(WRAPPER_FILE ${WRAPPED_FILES})
+    if(NOT EXISTS "${WRAPPER_FILE}")
+        file(WRITE ${WRAPPER_FILE} "")
+    endif()
+endforeach()
+    
+# Set up the project for building the code generated by shiboken.
+ly_add_target(
+    NAME ${BINDINGS_LIBRARY} SHARED
+    NAMESPACE Gem
+    FILES_CMAKE
+        wrapped_files.cmake
```

Comment:
This isn't the proper way to add a dependency on [generated](https://cmake.org/cmake/help/latest/prop_sf/GENERATED.html#prop_sf:GENERATED) files.

What could be done is to take the list of files in `wrapped_files.cmake` and added it as OUTPUT to the CUSTOM_COMMAND above.

Then that list of wrapped files can be added via the `target_sources` command to the Bindings Library target.

An example of doing this is in the [AzAutoGen](https://github.com/o3de/o3de/blob/development/cmake/LyAutoGen.cmake#L51) command.

Reply:
Fixed